### PR TITLE
Feature/migl/rdphoen 1140 redundand uboot env support

### PR DESCRIPTION
**Pull request by linkjumper**
This PR enables redundant storage of u-boot environment variables, as well as Linux access to the variables with fw_printenv/fw_setenv (libubootenv-bin).
- tested for imx8mpevk and imx8mp-irma6r2 with maintenance image

Jenkins IRMA6-Release 1: [#1280](http://jenkinsmaster.irisgmbh.local:8080/job/RD-V/job/IRMA6/view/Irma6_Yocto_Build_Firmware_Pipeline.feature/)

**How to build?**
```
$ bitbake mc:imx8mp-evk:irma6-maintenance-uuu
```

**How to test?**
Check rootfs (irma6-maintenance-uuu) for these files:
- /etc/u-boot-initial-env present?
- /etc/fw_env.config present?
- Tools lvm2, cryptsetup, fw_setenv/fw_printenv (libubootenv-bin) present?

```
# flash board
$ cd build/tmp/deploy/images/imx8mpevk/irma6-maintenance-uuu
$ uuu flashall.uuu
```

```
# With a freshly flashed sensor from Linux u-boot-env using /etc/u-boot-initial-env
# Save u-boot-env in Linux and read back in u-boot:
root@imx8mpevk:~# fw_printenv                                                   
Cannot read environment, using default   <-- ok
[..]
root@imx8mpevk:~# fw_setenv my_fancy_test 77                                    
Cannot read environment, using default
root@imx8mpevk:~# reboot
u-boot=> printenv my_fancy_test                                                 
my_fancy_test=77
```

```
# Store redundant envs in u-boot?
u-boot=> setenv my_fancy_test 12                                                
u-boot=> saveenv                                                                
Saving Environment to MMC... Writing to MMC(2)... OK                            
u-boot=> saveenv                                                                
Saving Environment to MMC... Writing to redundant MMC(2)... OK                  
u-boot=> saveenv                                                                
Saving Environment to MMC... Writing to MMC(2)... OK                            
u-boot=> saveenv                                                                
Saving Environment to MMC... Writing to redundant MMC(2)... OK                  
u-boot=> printenv my_fancy_test                                                 
my_fancy_test=12 
```

```
# Save u-boot-env in u-boot and read it in Linux:
u-boot=> boot
[..]
root@imx8mpevk:~# fw_printenv my_fancy_test                                     
my_fancy_test=12
```

### CODE REVIEW

File: recipes-support/irma6-swuimage/irma6-swuimage.inc [hunk 1]
```diff
@@ -10,6 +10,12 @@ IMAGENAME = "${@ d.getVar('PN').replace('-swuimage','')}"
 # The following line forces the usage of the "files" directory next to this .inc file even if it is required from another layer
 FILESEXTRAPATHS_prepend := "${@os.path.dirname(d.getVar('FILE'))}/files:"
 
+RDEPENDS_${PN} += " \
```
Comment (iris-ersc) [inline recipes-support/irma6-swuimage/irma6-swuimage.inc:13]: Please don't `rdepend` in this image recipe. The swuimage has no "runtime". A runtime dependency would basically be an `IMAGE_INSTALL` which should already be covered with the changes to `recipes-core/images/irma6-base.bb`

**Review by iris-ersc — CHANGES_REQUESTED** [review]

File: recipes-support/irma6-swuimage/irma6-swuimage.inc (hunk 1 above)
Comment (linkjumper) [reply recipes-support/irma6-swuimage/irma6-swuimage.inc]: The commit is dropped!

**Review by Eximmius — APPROVED** [review]
Tested with dev-uuu

**Review by sven-foerster — APPROVED** [review]
tested: irma6-maintenance-uuu

**Review by iris-ersc — APPROVED** [review]

**Comment by iris-ersc** [comment]
## Maintainer Checklist
- Is there a changelog entry? *Yes*
- Do the commit messages comply with the specifications? *Yes*
- Was a Jenkins run performed and was it error-free? *Yes*
- Is a manual rebase necessary? *no*

**Pull request merged**